### PR TITLE
tls: fix daysUntilFirstCertExpires() return nil when expired

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Loong Dai <loong.dai@intel.com>

Add days_until_first_cert_expiring stat integration test to ensure no nil error.

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by mattklein123** [comment]
There is no point in using an optional if you are going to return 0. Please fix the broken callsite to handle the lack of a value and add a test. Thanks!

/wait

File: test/config/integration/certs/expired_cert.pem [hunk 1]
```diff
@@ -0,0 +1,24 @@
+-----BEGIN CERTIFICATE-----
```
Comment (mattklein123) [inline test/config/integration/certs/expired_cert.pem:1]: Don't these files get generated by a script? Can you modify the script?

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Thanks for fixing and add the test. Please check CI also.

/wait

File: test/config/integration/certs/expired_cert.pem (hunk 1 above)
Comment (daixiang0) [reply test/config/integration/certs/expired_cert.pem:1]: Sure.

Comment (daixiang0) [reply test/config/integration/certs/expired_cert.pem:1]: Sorry about force push for DCO fix.

**Comment by daixiang0** [comment]
/retest

https://dev.azure.com/cncf/envoy/_build/results?buildId=109290&view=logs&j=8c169225-0ae8-53bd-947f-07cb81846cb5&t=6b0ace90-28b2-51d9-2951-b1fd35e67dec&l=5994

**Comment by daixiang0** [comment]
CI failed since https://dev.azure.com/cncf/envoy/_build/results?buildId=109300&view=logs&j=8c169225-0ae8-53bd-947f-07cb81846cb5&t=6b0ace90-28b2-51d9-2951-b1fd35e67dec&l=1406

**Comment by daixiang0** [comment]
https://dev.azure.com/cncf/envoy/_build/results?buildId=109369&view=logs&j=767be981-567e-57d8-68c3-2140ede0a0bd&t=1fb0b72f-6293-562d-a4d1-bb649223eaca&l=77
> Server terminated abruptly

**Comment by daixiang0** [comment]
@phlax seems something wrong with arm server:
https://dev.azure.com/cncf/envoy/_build/results?buildId=109369&view=logs&j=767be981-567e-57d8-68c3-2140ede0a0bd&t=1fb0b72f-6293-562d-a4d1-bb649223eaca&l=81

File: test/config/integration/certs/expired_cert_hash.h [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// NOLINT(namespace-envoy)
+constexpr char TEST_EXPIRED__CERT_HASH[] = "FC:F7:07:14:C3:0D:B4:BE:0B:BF:23:9B:C2:09:DA:CD:54:66:"
+                                           "32:65:07:50:35:E8:D0:14:ED:D6:B1:96:A1:3C";
```
Comment (mattklein123) [inline test/config/integration/certs/expired_cert_hash.h:3]: I think this is unused. Delete?

File: test/config/integration/certs/servercert_info.h [hunk 3]
```diff
@@ -0,0 +1,5 @@
+// NOLINT(namespace-envoy)
+constexpr char TEST_SERVER_CERT_HASH[] = "DC:E2:2B:65:90:43:9A:36:1C:8E:6D:CA:42:8A:8C:37:C7:A1:77:"
+                                         "00:5B:C1:3E:33:8A:B9:2D:04:2C:B1:3F:0A";
+constexpr char TEST_SERVER_CERT_NOT_BEFORE[] = "Apr 7 16:46:35 2022 GMT";
+constexpr char TEST_SERVER_CERT_NOT_AFTER[] = "Apr 6 16:46:35 2024 GMT";
```
Comment (mattklein123) [inline test/config/integration/certs/servercert_info.h:5]: Can you add comments on where this came from and how to regenerate it if needed?

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Thanks, LGTM with small comments. Sorry for the CI problems I think they are fixed.

/wait

File: test/config/integration/certs/expired_cert_hash.h (hunk 2 above)
Comment (daixiang0) [reply test/config/integration/certs/expired_cert_hash.h:3]: Now `certs.sh` will generate a hash file, not only here, but also in other tests, although some of the hash files are unused.

File: test/config/integration/certs/servercert_info.h (hunk 3 above)
Comment (daixiang0) [reply test/config/integration/certs/servercert_info.h:5]: All are generated by `certs.h`, refer to https://github.com/envoyproxy/envoy/blob/main/test/config/integration/certs/README.md, or I can update that script and add comments during generating.

File: test/config/integration/certs/expired_cert_hash.h (hunk 2 above)
Comment (mattklein123) [reply test/config/integration/certs/expired_cert_hash.h:3]: OK I see, that's fine, thanks.

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**